File: smoke_tests/php/sql_injection.php
<?php
// SQL Injection vulnerabilities in PHP

class SqlInjectionVulnerabilities {
    private $pdo;
    private $mysqli;

    public function getUserUnsafe($userId) {
        // VULNERABLE: String concatenation in SQL
        $query = "SELECT * FROM users WHERE id = '" . $userId . "'";
        return $this->pdo->query($query);
    }

    public function loginUnsafe($username, $password) {
        // VULNERABLE: SQL injection in login
        $sql = "SELECT * FROM users WHERE username='$username' AND password='$password'";
        return mysqli_query($this->mysqli, $sql);
    }

    public function searchUnsafe($term) {
        // VULNERABLE: SQL injection in search
        $query = "SELECT * FROM products WHERE name LIKE '%$term%'";
        return $this->pdo->query($query);
    }

    public function deleteUnsafe($tableName, $id) {
        // VULNERABLE: Table name injection
        $sql = "DELETE FROM $tableName WHERE id = $id";
        $this->pdo->exec($sql);
    }

    public function executeRaw($query) {
        // VULNERABLE: Direct query execution
        mysql_query($query);
    }

    public function unsafePrepare($column, $value) {
        // VULNERABLE: Column name from user input
        $sql = "SELECT * FROM users WHERE $column = ?";
        $stmt = $this->pdo->prepare($sql);
        $stmt->execute([$value]);
    }
}

File: smoke_tests/swift/sql_injection.swift
// SQL Injection vulnerabilities in Swift
import Foundation
import SQLite3

class SqlInjectionVulnerabilities {
    var db: OpaquePointer?

    func getUserUnsafe(userId: String) -> String {
        // VULNERABLE: String interpolation in SQL
        let query = "SELECT * FROM users WHERE id = '\(userId)'"
        var stmt: OpaquePointer?
        sqlite3_prepare_v2(db, query, -1, &stmt, nil)
        return ""
    }

    func loginUnsafe(username: String, password: String) -> Bool {
        // VULNERABLE: SQL injection in login
        let sql = "SELECT * FROM users WHERE username='\(username)' AND password='\(password)'"
        var stmt: OpaquePointer?
        sqlite3_prepare_v2(db, sql, -1, &stmt, nil)
        return sqlite3_step(stmt) == SQLITE_ROW
    }

    func searchUnsafe(term: String) -> [String] {
        // VULNERABLE: SQL injection in search
        let query = "SELECT * FROM products WHERE name LIKE '%\(term)%'"
        var stmt: OpaquePointer?
        sqlite3_prepare_v2(db, query, -1, &stmt, nil)
        return []
    }

    func deleteUnsafe(tableName: String, id: Int) {
        // VULNERABLE: Table name injection
        let sql = "DELETE FROM \(tableName) WHERE id = \(id)"
        sqlite3_exec(db, sql, nil, nil, nil)
    }

    func executeRaw(query: String) {
        // VULNERABLE: Direct query execution
        sqlite3_exec(db, query, nil, nil, nil)
    }
}
